packages/core/test/model.test.ts

```diff
@@ -220,4 +220,19 @@ describe('createModel', () => {
 
     expect(machine.initialState.context.count).toBe(0);
   });
+
+  it('should not compile if missing context', () => {
+    const toggleModel = createModel({ count: 0 });
+
+    // @ts-expect-error
+    const m = createMachine<typeof toggleModel>({
+      id: 'machine',
+      initial: 'inactive',
+      // missing context:
+      // context: toggleModel.initialContext,
```

Comment:
A question that I feel is somewhat related to this.

There are cases for which the initial context is not known upfront - for example when the machine requires user data but it can only be passed from the parent since it first needs to be acquired from somewhere. With this change, you force a user to provide some dummy data here or you steer them towards casting which loses the type safety. Of course, currently, there is no type safety when you don't provide any context value, and yet you expect context to be of a certain type but that is an orthogonal problem. And it has existed before the model has even been introduced.

So the question is - what the user should do when using modal for cases like the one that I've described. What's the official recommendation?

Reply:
With this pattern, that's not an issue:

```ts
const createUserMachine = user => createMachine({
  context: { user }
});
```

Reply:
Another idea (from Discord) for reference:

```js
// TENTATIVE API
const machine = createMachine({
  // input is of type Partial<TContext> maybe
  context: (input) => ({
    some: 'default value',
    ...input
  })
});

machine.transition(/* ... */); // throws error; expecting input

machine.withContext({ the: 'input' }).transition(/* ... */);

interpret(machine); // throws error

interpret(machine.withContext({ ... })); // does not throw error
```